akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -72,14 +72,14 @@ private[affinity] object AffinityPool {
     private[this] var parkPeriodNs = 0L
     @volatile private[this] var idling = false
 
-    @inline private[this] final def transitionTo(newState: IdleState): Unit = {
+    @inline private[this] def transitionTo(newState: IdleState): Unit = {
       state = newState
       turns = 0
     }
 
-    final def isIdling: Boolean = idling
+    def isIdling: Boolean = idling
```

Comment:
As far as I know hotspot doesn't care about `final` for methods and classes so this change should be ok. @viktorklang have you looked into the details of if `final` for methods and classes matters for performance?

Reply:
I don't see the point in removing the `final` modifier. if a class is made non-final then how does that person know what fields should now be final?

Reply:
ok, that is not a strong argument in my opinion. removing `final` on a public class shouldn't be taken lightly because it can open up a whole new set of compatibility considerations. Trusting that the previous author had sprinkled final at all places that matters would not be enough. Then all methods should be made final, since they effectively were that when the class was final.

Here it's not about public API, so even less important. I was only curious if there were any performance considerations.

Reply:
Yeah, I prefer all things that can be final made final so that there is no question about intent.

Reply:
yeah, that will not work with a shared big code base like Akka, since that is not common practice (it is perceived as redundant noise). When reviewing contributions we would do nothing but reminding everybody (including ourselves) to add final everywhere, and we would anyway forget it in places (so can't be trusted).